src/java_binding/java/java-binding/src/main/java/com/risingwave/java/binding/Iterator.java

```diff
@@ -0,0 +1,27 @@
+package com.risingwave.java.binding;
+
+public class Iterator implements AutoCloseable {
+    final long pointer;
+    boolean isClosed;
+
+    public Iterator() {
```

Comment:
I think this is a overly-simplified version of the iterator constructor since we didn't pass any storage related information to it such as version, epoch, key range, etc. Same for `Binding.iteratorNew()`. I am thinking whether we should rename `iteratorNew` to something like `iteratorNewForDemo` to make it clearer that the current interfaces are implemented only for demo purpose. What do you think?

Reply:
I think we can keep the naming `iteratorNew`. The current implementation is only for demo purpose because the data returned from the iterator is mocked, and therefore has nothing storage related config to pass. But the semantic of this method is to return a normal iterator.

In future PR https://github.com/risingwavelabs/risingwave/pull/7253, we will pass the storage related config to `iteratorNew`.

Reply:
> I think we can keep the naming `iteratorNew`. The current implementation is only for demo purpose because the data returned from the iterator is mocked, and therefore has nothing storage related config to pass. But the semantic of this method is to return a normal iterator.
> 
> In future PR #7253, we will pass the storage related config to `iteratorNew`.

Cool. Sounds good to me.